qiskit/providers/backend.py

```diff
@@ -229,3 +239,390 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class QubitProperties:
+    """A representation of the properties of a qubit on a backend.
+
+    This class provides the optional properties that a backend can provide for
+    a qubit. These represent the set of qubit properties that Qiskit can
+    currently work with if present. Howeverm if your backend provides additional
+    properties of qubits you should subclass this to add additional custom
+    attributes for those custom/additional properties provided by the backend.
+    """
+
+    __slots = ("t1", "t2", "frequency")
+
+    def __init__(self, t1=None, t2=None, frequency=None, properties=None):
+        """Create a new ``QubitProperties`` object
+
+        Args:
+            t1: The T1 time for a qubit in seconds
+            t2: The T2 time for a qubit in seconds
+            frequency: The frequency of a qubit in Hz
+            properties: A free form dictionary of additional properties the
+                backend has for a qubit.
+        """
+        self.t1 = t1
+        self.t2 = t2
+        self.frequency = frequency
+        self.properties = properties
+
+    def __repr__(self):
+        return (
+            f"QubitProperties(t1={self.t1}, t2={self.t2}, "
+            f"frequency={self.frequency}, properties={self.properties})"
+        )
+
+
+class BackendV2(Backend, ABC):
+    """Abstract class for Backends
+
+    This abstract class is to be used for all Backend objects created by a
+    provider. This version differs from earlier abstract Backend classes in
+    that the configuration attribute no longer exists instead attributes
+    exposing equivalent required immutable properties of the backend device
```

Comment:
What does "immutable" mean in this case? Cannot change with out e.g. a device revision?

Reply:
I just mean there is no setter on the property

Reply:
oh on re-reading this. I meant things like the coupling map or number of qubits. So yeah things that can't change without a device revision.